From 3ec2c4571f701c61d157e40dcebbfef43a38c963 Mon Sep 17 00:00:00 2001
From: Hunter Laux <hunterlaux@gmail.com>
Date: Wed, 6 Apr 2016 00:54:05 -0700
Subject: [PATCH 09/13] usb: phy: tegra: Add 38.4MHz clock table entry

The Tegra210 uses a 38.4MHz OSC. This clock table entry is required to
use the ehci phy on the Jetson TX1.

The xtal_freq_count is actually a 12 bit value, so it should be a u16
instead of u8.

Signed-off-by: Hunter Laux <hunterlaux@gmail.com>
---
 drivers/usb/phy/phy-tegra-usb.c | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/drivers/usb/phy/phy-tegra-usb.c b/drivers/usb/phy/phy-tegra-usb.c
index 68cd4b68e3a2..b36af02eddee 100644
--- a/drivers/usb/phy/phy-tegra-usb.c
+++ b/drivers/usb/phy/phy-tegra-usb.c
@@ -174,7 +174,7 @@ struct tegra_xtal_freq {
 	u8 enable_delay;
 	u8 stable_count;
 	u8 active_delay;
-	u8 xtal_freq_count;
+	u16 xtal_freq_count;
 	u16 debounce;
 };
 
@@ -211,6 +211,14 @@ static const struct tegra_xtal_freq tegra_freq_table[] = {
 		.xtal_freq_count = 0xFE,
 		.debounce = 0xFDE8,
 	},
+	{
+		.freq = 38400000,
+		.enable_delay = 0x00,
+		.stable_count = 0x00,
+		.active_delay = 0x18,
+		.xtal_freq_count = 0x177,
+		.debounce = 0xBB80,
+	},
 };
 
 static inline struct tegra_usb_phy *to_tegra_usb_phy(struct usb_phy *u_phy)
-- 
2.35.1

